pydra/engine/task.py

```diff
@@ -275,15 +276,32 @@ def __init__(
             TODO
 
         """
-        if input_spec is None:
-            input_spec = SpecInfo(name="Inputs", fields=[], bases=(ShellSpec,))
-        self.input_spec = input_spec
```

Comment:
this is set in the init

Reply:
yes, the old code was working fine.

But what I did in this PR is allowing to use the class attribute (if no `input_spec` is provided during initialization), so it can use the default `input_spec` for the class, se [this](https://github.com/nipype/pydra/pull/348/files#diff-1a747d8c932de194d2dda44683c02558R22).

Reply:
so before I set `input_spec = SpecInfo(name="Inputs", fields=[], bases=(ShellSpec,))`, I also want to check for the class attribute

Reply:
you can do something like this:

```
In [12]: class A:
    ...:     var1 = None
    ...:     def __init__(self, var1=None):
    ...:         self.var1 = var1 or self.var1 or 'fixed'
    ...: 

In [13]: a = A()

In [14]: a.__dict__
Out[14]: {'var1': 'fixed'}

In [15]: class B(A):
    ...:     var1 = 'derived'
    ...: 

In [16]: b = B()

In [17]: b.__dict__
Out[17]: {'var1': 'derived'}

```

Reply:
sure, i believe this is exactly what i did, just used more explicit syntax, but can move to yours.

Reply:
it's definitely much shorter :)